### PR TITLE
Invite sub-org users to add email/password creds

### PR DESCRIPTION
**Pull request by hari-hypto**
Changes
1. Support to create a user with no login access using "/users" API
2. Invite user to attach username/password credentials using "/verifyEmail" API
3. Create password for the invited user using "/create_password" API.

Test:
UTs added

### CODE REVIEW

File: src/main/kotlin/com/hypto/iam/server/service/UsersService.kt [hunk 1]
```diff
@@ -159,12 +159,15 @@ class UsersServiceImpl : KoinComponent, UsersService {
     ) {
         val org = organizationRepo.findById(organizationId)
             ?: throw EntityNotFoundException("Invalid organization id")
+        val encodedEmail = email?.let {
+            getEmail(organizationId, subOrganizationName, email)
+        } ?: throw BadRequestException("Email is required")
```
Comment (karthick-vinod) [inline src/main/kotlin/com/hypto/iam/server/service/UsersService.kt:164]:
```suggestion
        val encodedEmail = getEmail(organizationId, subOrganizationName, email) ?: throw BadRequestException("Email is required")
```

File: src/main/kotlin/com/hypto/iam/server/validators/RequestModelValidator.kt [hunk 2]
```diff
@@ -165,6 +166,12 @@ fun ResetPasswordRequest.validate(): ResetPasswordRequest {
 
 fun VerifyEmailRequest.validate(): VerifyEmailRequest {
     verifyEmailRequestValidation.validateAndThrowOnFailure(this)
+    Validation {
+        (this::oneOf)(
+            this@validate,
+            listOf(VerifyEmailRequest::email, VerifyEmailRequest::userHrn)
+        )
```
Comment (karthick-vinod) [inline src/main/kotlin/com/hypto/iam/server/validators/RequestModelValidator.kt:173]: Can this be moved inside `verifyEmailRequestValidation`?

**Review by karthick-vinod — APPROVED** [review]

File: src/main/kotlin/com/hypto/iam/server/service/UsersService.kt (hunk 1 above)
Comment (hari-hypto) [reply src/main/kotlin/com/hypto/iam/server/service/UsersService.kt]: The problem here is the parameter "email" is non nullable in getEmail method, so this nullable check is required.

File: src/main/kotlin/com/hypto/iam/server/validators/RequestModelValidator.kt (hunk 2 above)
Comment (hari-hypto) [reply src/main/kotlin/com/hypto/iam/server/validators/RequestModelValidator.kt:173]: As discussed, this is not required. So resolving.

**Pull request merged**